### ISSUE TITLE
Results with public checkpoints

### ISSUE THREAD

**Issue by YovaKem** [ufal/augpt#6]
Running the evaluation in the following fashion:
```
./generate.py --model jkulhanek/augpt-mw-21 --dataset multiwoz-2.1-test --file predictions.txt
./evaluate_multiwoz.py --file predictions.txt --dataset multiwoz-2.1-test
```
I get inform and success results somewhat lower than what is reported in the paper (Table 1):

> info: match: 0.8440, success: 0.6780
info: computing bleu
info: test bleu: 0.0000
info: test delex bleu: 0.1732

Is this to be expected? Is the `jkulhanek/augpt-mw-21 ` checkpoint the same one used to obtain the results in Table 1?

**Comment by jkulhanek** [comment]
There was a bug in evaluation in the original source code. We have a new version of the paper under submission.

**Comment by YovaKem** [comment]
Ok, I see. But the code is now up to date and I can assume I'm getting the right results?

**Comment by jkulhanek** [comment]
Yes, the results are consistent with the original evaluation published by Budzianowski.

**Comment by YovaKem** [comment]
Thanks. And can I ask for a clarification on why responses are [not lexicalized](https://github.com/ufal/augpt/blob/fa8a57961ed1d8fe6099978c489c0b0f8956d64e/generate.py#L168) in `generate.py`? Does that also have to do with keeping the evaluation procedure consistent across works?

**Comment by jkulhanek** [comment]
The generate script generates delexicalized responses to be evaluated by the MultiWOZ evaluation (which uses delexicalized responses). You can change [this line](https://github.com/ufal/augpt/blob/fa8a57961ed1d8fe6099978c489c0b0f8956d64e/generate.py#L168) or use interact.py script.

**Comment by YovaKem** [comment]
Thanks. I noticed one small thing while using the `generate.py` code - there is some randomness in the predictions made, which can be avoided by fixing the seed.

May I also ask whether it is standard in evaluation of dialog systems that when a new system response is predicted, it is conditioned on past *gold standard* system responses rather than past generated system responses? This appears to be the case in `generate.py`, but to me it seems more "fair" let's say, or more realistic to condition the generation on past *generated*  system utterances.

**Comment by YovaKem** [comment]
@jkulhanek , I also ran the evaluation script [here](https://github.com/Tomiinek/MultiWOZ_Evaluation/blob/master/evaluate.py) (which is the official recommendation as stated [here](https://github.com/budzianowski/multiwoz)) and obtained the following results for the predictions of `jkulhanek/augpt-mw-21`:

> ====== BLEU ======
> damd           16.71749836889923
> uniconv        15.605802335898408
> hdsa           15.56793789619695
> lava           15.539192586808422
> augpt          17.242521108846592
> mwz22          17.013728423986837
> 
> ====== SUCCESS ======
> inform         {'total': 75.5, 'taxi': 100.0, 'attraction': 85.6, 'restaurant': 83.1, 'train': 96.6, 'hotel': 72.8}
> success        {'total': 60.9, 'taxi': 68.7, 'attraction': 56.6, 'restaurant': 64.5, 'train': 78.2, 'hotel': 60.2}

Do you know where the mismatch in inform and success stems from?

**Comment by jkulhanek** [comment]
@Tomiinek  can you please look into it?

**Comment by Tomiinek** [comment]
Hello @YovaKem,

First, I have to say that I am not completely sure guys :slightly_smiling_face: There might be multiple factors.

The difference between the current and the very first numbers were caused by the approach to *Venue updates* (see the [evaluation paper](https://arxiv.org/pdf/2106.05555.pdf)). I believe that the current difference is caused mainly by the method of *Venue comparison*, see [these lines](https://github.com/ufal/augpt/blob/fa8a57961ed1d8fe6099978c489c0b0f8956d64e/data/evaluation/multiwoz/evaluator.py#L111-L112) and the paper for more details.

The current AuGPT venue comparison (checking whether there exists *any* overlap between predicted and ground truth database results) is probably more compatible with the method used in DAMD, MinTL, etc., but the right and hopefully preffered method is the one used, for example, in MarCo (see [these lines](https://github.com/InitialBug/MarCo-Dialog/blob/2b9ff5929e4b924ec130cd7ceeeda9991186ab99/evaluator.py#L184)), i.e. checking whether the database results corresponding to current dialogue state are a non-empty subset of the ground-truth results.   

Hope it helps :innocent:

**Comment by YovaKem** [comment]
Hi @Tomiinek, thanks for the pointers. I'll look into those! :)
